strax/plugin.py

```diff
@@ -847,3 +847,50 @@ def cleanup(self, iters, wait_for):
             for s in savers:
                 s.close(wait_for=wait_for)
         super().cleanup(iters, wait_for)
+
+
+@export
+class CutPlugin(Plugin):
+    """Generate a plugin that provides a boolean for a given cut specified by 'cut_by'"""
+    save_when = SaveWhen.NEVER
+
+    def __init__(self):
+        super().__init__()
+
+        _name = strax.camel_to_snake(self.__class__.__name__)
+        if not hasattr(self, 'provides'):
+            self.provides = _name
+        if not hasattr(self, 'cut_name'):
+            self.cut_name = _name
+        if not hasattr(self, 'cut_description'):
+            _description = _name
+            if 'cut_' not in _description:
+                _description = 'Cut by ' + _description
+            else:
+                _description = " ".join(_description.split("_"))
+            self.cut_description = _description
+
+    def infer_dtype(self):
+        dtype = [(self.cut_name, np.bool, self.cut_description)]
+        dtype = dtype + strax.time_fields
+        return dtype
+
+    def compute(self, **kwargs):
```

Comment:
To get the start and endtime of your data you can do 

`compute(<data_kinds>, start, end)` 

See: https://github.com/AxFoundation/strax/blob/7c0641000301653adbf4144967b8a9cc18bf8103/strax/plugin.py#L448 and my comment below.

Reply:
I will use strax.endtime as it is somewhat easier